contracts/SSVNetwork.sol

```diff
@@ -627,16 +629,15 @@ contract SSVNetwork is UUPSUpgradeable, Ownable2StepUpgradeable, ISSVNetwork {
         if (operator.owner != msg.sender) revert CallerNotOwner();
     }
 
-    function _registerValidatorPublicKey(bytes calldata publicKey) private {
-        if (publicKey.length != 48) {
-            revert InvalidPublicKeyLength();
-        }
+    function _registerValidatorPublicKey(bytes calldata publicKey, bytes32 hashedOpIds) private {
+        if (publicKey.length != 48) revert InvalidPublicKeyLength();
 
-        bytes32 hashedPk = keccak256(publicKey);
-        if (validatorPKs[hashedPk].owner != address(0)) {
+        bytes32 hashedPk = keccak256(abi.encodePacked(publicKey, msg.sender));
+
+        if (validatorPKs[hashedPk].hashedOperatorIds == hashedOpIds) {
```

Comment:
This is wrong, we should allow only one validator per user and not per user & cluster

Reply:
also need to add a test to enforce it

Reply:
resolved by @mtabasco